js/tests/unit/modal.js

```diff
@@ -697,4 +697,50 @@ $(function () {
 
     beginTimestamp = Date.now()
   })
+
+  QUnit.test('should dispose modal', function (assert) {
+    assert.expect(2)
+    var done = assert.async()
+
+    var $modal = $([
+      '<div id="modal-test">',
+      '  <div class="modal-dialog">',
+      '    <div class="modal-content">',
+      '      <div class="modal-body" />',
+      '    </div>',
+      '  </div>',
+      '</div>'
+    ].join('')).appendTo('#qunit-fixture')
+
+    $modal.on('shown.bs.modal', function () {
+      var data = $(this).data()
+      var element = data['bs.modal']._element
+      var dialog = data['bs.modal']._dialog
+
+      $(this).bootstrapModal('dispose')
+
+      assert.ok(typeof data['bs.modal'] === 'undefined', 'modal data object was disposed')
+
+      const modalEvents = [window, document, element, dialog]
```

Comment:
Instead of doing that, you should use SinonJS spy to be sure we call `.off` on those elements